### ISSUE TITLE
unrecognized command line option ‘-stdlib=libc++’ when making with autogen on On Mac OS X 10.11.3

### ISSUE THREAD

**Issue by kamal94** [JohnLangford/vowpal_wabbit#965]
``` git clone https://github.com/JohnLangford/vowpal_wabbit ```
``` cd vowpal_wabbit ```
``` sh autogen.sh --enable-libc++ ```
``` make ```

OUTPUT:

git clone https://github.com/JohnLangford/vowpal_wabbit
Cloning into 'vowpal_wabbit'...
remote: Counting objects: 38776, done.
remote: Total 38776 (delta 0), reused 0 (delta 0), pack-reused 38776
Receiving objects: 100% (38776/38776), 97.58 MiB | 3.22 MiB/s, done.
Resolving deltas: 100% (29241/29241), done.
Checking connectivity... done.

 18:36:23  ~/Repositories 
$ cd vowpal_wabbit 

 18:36:37  ~/Repositories/vowpal_wabbit    master ✔ 
$ sh autogen.sh --enable-libc++
glibtoolize: putting auxiliary files in '.'.
glibtoolize: copying file './ltmain.sh'
glibtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'acinclude.d'.
glibtoolize: copying file 'acinclude.d/libtool.m4'
glibtoolize: copying file 'acinclude.d/ltoptions.m4'
glibtoolize: copying file 'acinclude.d/ltsugar.m4'
glibtoolize: copying file 'acinclude.d/ltversion.m4'
glibtoolize: copying file 'acinclude.d/lt~obsolete.m4'
configure.ac:15: installing './compile'
configure.ac:13: installing './config.guess'
configure.ac:13: installing './config.sub'
configure.ac:3: installing './install-sh'
configure.ac:3: installing './missing'
cluster/Makefile.am: installing './depcomp'
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking for style of include used by make... GNU
checking dependency style of g++... gcc3
checking how to run the C++ preprocessor... g++ -E
checking build system type... x86_64-apple-darwin15.3.0
checking host system type... x86_64-apple-darwin15.3.0
checking how to print strings... printf
checking for gcc... gcc
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm
checking the name lister (/usr/bin/nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 196608
checking how to convert x86_64-apple-darwin15.3.0 file names to x86_64-apple-darwin15.3.0 format... func_convert_file_noop
checking how to convert x86_64-apple-darwin15.3.0 file names to toolchain format... func_convert_file_noop
checking for /usr/bin/ld option to reload object files... -r
checking for objdump... no
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... no
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fno-common -DPIC
checking if gcc PIC flag -fno-common -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin15.3.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... no
checking whether the g++ linker (/usr/bin/ld) supports shared libraries... yes
checking for g++ option to produce PIC... -fno-common -DPIC
checking if g++ PIC flag -fno-common -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin15.3.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether g++ supports C++11 features by default... no
checking whether g++ supports C++11 features with -std=gnu++11... yes
checking for ANSI C header files... (cached) yes
checking for boostlib >= 1.0... yes
checking whether the Boost::Program_Options library is available... yes
checking for exit in -lboost_program_options-mt... yes
checking if zlib is wanted... yes
checking for pthread_join in LIBS=-lpthread with CFLAGS=... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking if more special flags are required for pthreads... -D_THREAD_SAFE
checking for PTHREAD_PRIO_INHERIT... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating vowpalwabbit/Makefile
config.status: creating cluster/Makefile
config.status: creating library/Makefile
config.status: creating libvw.pc
config.status: creating libvw_c_wrapper.pc
config.status: creating vowpalwabbit/config.h
config.status: executing depfiles commands
config.status: executing libtool commands

 18:37:10  ~/Repositories/vowpal_wabbit    master ✘ ✹ 
$ make
Making all in vowpalwabbit
/Library/Developer/CommandLineTools/usr/bin/make  all-am
depbase=`echo allreduce_sockets.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
	/bin/sh ../libtool  --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -I/usr/local/include -I/include -D_THREAD_SAFE  -Wall -Wno-unused-local-typedefs    -pedantic -stdlib=libc++ -ffast-math   -O3 -fomit-frame-pointer -DNDEBUG -std=gnu++11 -MT allreduce_sockets.lo -MD -MP -MF $depbase.Tpo -c -o allreduce_sockets.lo allreduce_sockets.cc &&\
	mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -I/usr/local/include -I/include -D_THREAD_SAFE -Wall -Wno-unused-local-typedefs -pedantic -stdlib=libc++ -ffast-math -O3 -fomit-frame-pointer -DNDEBUG -std=gnu++11 -MT allreduce_sockets.lo -MD -MP -MF .deps/allreduce_sockets.Tpo -c allreduce_sockets.cc  -fno-common -DPIC -o .libs/allreduce_sockets.o
g++: error: unrecognized command line option ‘-stdlib=libc++’
make[2]: *** [allreduce_sockets.lo] Error 1
make[1]: *** [all] Error 2
make: *** [all-recursive] Error 1

**Comment by JohnLangford** [comment]
Have you tried using the default Makefile?

**Comment by kamal94** [comment]
I believe this was caused because I was using the gcc (g++) compiler. 
[hint 1](http://stackoverflow.com/questions/24419832/unrecognized-command-line-option-stdlib-libc-with-macports-gcc48)
[hint 2](http://stackoverflow.com/questions/16352833/linking-with-clang-on-os-x-generates-lots-of-symbol-not-found-errors)
It was solved by running 
```./configure --enable-libc++ CXX=clang++```
after ```sh autogen.sh --enable-libc++```
and then running ```make```

**solution workflow**:
```
git clone https://github.com/JohnLangford/vowpal_wabbit
cd vowpal_wabbit
sh autogen.sh --enable-libc++
./configure --enable-libc++ CXX=clang++
make
```

**By the way: This was only solved after the fix in #971.**

Also, it seems that passing --enable-libc++ CXX=clang++ to sh autogen.sh does not have the same effect as passing it to ./configure. I am not sure if that is intentional, but if they should be producing similar results, then that is not happening.
